{
    "cells": [
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": "\n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "knitr::opts_chunk$set(echo = TRUE)\n",
                "knitr::opts_chunk$set(warning = FALSE, message = FALSE) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "### R code used to generate figures for the paper: Signatures of landscape and captivity in the gut microbiota of Southern Hairy-nosed Wombats (Lasiorhinus latifrons)\n",
                "\n",
                "## Load libraries\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "library(phyloseq)\n",
                "library(qiime2R)\n",
                "library(dplyr)\n",
                "library(svglite)\n",
                "library(cowplot)\n",
                "library(gplots)\n",
                "library(ggplot2)\n",
                "library(scales)\n",
                "library(ggpubr)\n",
                "library(tidyr)\n",
                "library(knitr)\n",
                "library(microbiome)\n",
                "library(microbiomeutilities)\n",
                "library(VennDiagram)\n",
                "library(ggVennDiagram)\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "## Import data\n",
                "\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "ps <- qza_to_phyloseq(\n",
                "  features = \"QIIME2_outputs/SHNW-gut-table-final.qza\",\n",
                "  tree = \"QIIME2_outputs/sepp-tree.qza\",\n",
                "  taxonomy = \"QIIME2_outputs/SHNW-gut-SILVA-132.qza\",\n",
                "  metadata = \"QIIME2_outputs/SHNW_2019_Gut_Metadata.txt\"\n",
                ")\n",
                "\n",
                "#PCoA matrices\n",
                "pcoa_uwUniFrac_all <- read_qza(\"QIIME2_outputs/SHNW-gut-Core-metrics-final-filtered-table-36346/unweighted_unifrac_pcoa_results.qza\")\n",
                "pcoa_wUniFrac_all <- read_qza(\"QIIME2_outputs/SHNW-gut-Core-metrics-final-filtered-table-36346/weighted_unifrac_pcoa_results.qza\")\n",
                "pcoa_uwUniFrac_wild <- read_qza(\"QIIME2_outputs/SHNW-gut-Core-metrics-final-filtered-table-36346-WILDonly/unweighted_unifrac_pcoa_results.qza\")\n",
                "pcoa_wUniFrac_wild <- read_qza(\"QIIME2_outputs/SHNW-gut-Core-metrics-final-filtered-table-36346-WILDonly/weighted_unifrac_pcoa_results.qza\")\n",
                "\n",
                "#Alpha divresity data\n",
                "faithsPD <- read.csv(\"QIIME2_outputs/faithsPD.tsv\", sep = '\\t')\n",
                "ObsFeat <- read.csv(\"QIIME2_outputs/observed_otus.tsv\", sep = '\\t')\n",
                "\n",
                "#Venn diagram data (skip second column, which is frequency of ASVs)\n",
                "captive.asvs <- read.csv(\"QIIME2_outputs/feature-frequency-detail-captive-200.csv\", header = FALSE, row.names = 1)\n",
                "wild.asvs <- read.csv(\"QIIME2_outputs/feature-frequency-detail-wild-750.csv\", header = FALSE, row.names = 1)\n",
                "\n",
                "brookfield.asvs <- read.csv(\"QIIME2_outputs/feature-frequency-detail-Brookfield.csv\", header = FALSE, row.names = 1)\n",
                "\n",
                "kooloola.asvs <- read.csv(\"QIIME2_outputs/feature-frequency-detail-Kooloola.csv\", header = FALSE, row.names = 1)\n",
                "\n",
                "wonga.asvs <- read.csv(\"QIIME2_outputs/feature-frequency-detail-Wonga.csv\", header = FALSE, row.names = 1)\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "## Alpha diversity plots (figure 2):\n",
                "\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "#This sets the order in which you wish to display the samples\n",
                "level_order_adiv <- c('Kooloola', 'Brookfield', 'Wonga', 'Captive')\n",
                "\n",
                "#Plot the data!\n",
                "sp_adiv_fig_faithsPD <- ggplot(faithsPD, aes(x = factor(Population, level=level_order_adiv), y=faith_pd))\n",
                "sp_adiv_fig_ObsFeat <- ggplot(ObsFeat, aes(x = factor(Population, level=level_order_adiv), y=observed_otus))\n",
                "\n",
                "#Colour vision deficiency-friendly palette\n",
                "cbp2_adiv <- c(\"orange\", \"black\", \"red\", \"blue\")\n",
                "\n",
                "#Faith's PD\n",
                "#add jitterm, size, and colour\n",
                "sp_adiv_fig_faithsPD + \n",
                "  #Boxplot\n",
                "  geom_boxplot(size=2, outlier.shape=4, outlier.size=3, outlier.stroke=3, aes(colour=Population)) +\n",
                "  #Jitter, size, colour\n",
                "  geom_jitter(position=position_dodge2(0.3), size=5, aes(colour=Population)) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_adiv) +\n",
                "  #Tick labels\n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=20), \n",
                "        axis.text.y = element_text(face=\"bold\", size=20),\n",
                "        axis.title.x = element_text(size=24, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=24, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.position = \"none\") +\n",
                "  #legend.title = element_text(size=0),\n",
                "  #legend.text = element_text(size=0),\n",
                "  #legend.key = element_rect(fill = \"white\", color = NA),\n",
                "  #legend.key.size = unit(0, \"line\")) +\n",
                "  #Axis labels\n",
                "  scale_y_continuous(breaks=seq(0,400,25)) +\n",
                "  labs(x = \"\\nPopulation\") +\n",
                "  labs(y = \"Faith's Phylogenetic Diversity\\n\")\n",
                "\n",
                "#Save image as .svg\n",
                "ggsave(filename = \"Fig_AlphaDiv-FaithsPD.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "\n",
                "#Observed OTUs\n",
                "#add jitterm, size, and colour\n",
                "sp_adiv_fig_ObsFeat + \n",
                "  #Boxplot\n",
                "  geom_boxplot(size=2, outlier.shape=4, outlier.size=3, outlier.stroke=3, aes(colour=Population)) +\n",
                "  #Jitter, size, colour\n",
                "  geom_jitter(position=position_dodge2(0.3), size=5, aes(colour=Population)) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_adiv) +\n",
                "  #Tick labels\n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=20), \n",
                "        axis.text.y = element_text(face=\"bold\", size=20),\n",
                "        axis.title.x = element_text(size=24, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=24, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.position = \"none\") +\n",
                "  #legend.title = element_text(size=0),\n",
                "  #legend.text = element_text(size=0),\n",
                "  #legend.key = element_rect(fill = \"white\", color = NA),\n",
                "  #legend.key.size = unit(0, \"line\")) +\n",
                "  #Axis labels\n",
                "  scale_y_continuous(breaks=seq(0,1000,100)) +\n",
                "  labs(x = \"\\nPopulation\") +\n",
                "  labs(y = \"Observed features\\n\")\n",
                "\n",
                "#Save image as .svg\n",
                "ggsave(filename = \"Fig_AlphaDiv-ObservedFeatures.svg\", width = 20, height = 11, dpi = 300)\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "## Beta diversity PCoA plots (figures 2 and 4):\n",
                "\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "#Load metadata\n",
                "metadata_pcoa <- read.csv(\"QIIME2_outputs/SHNW_2019_Gut_Metadata.txt\", sep = '\\t')\n",
                "\n",
                "#Colour palette\n",
                "cbp2_beta <- c(\"orange\", \"black\", \"red\", \"blue\")\n",
                "\n",
                "#Plot PC1 vs. PC2 (uwUniFrac_all)\n",
                "pcoa_uwUniFrac_all$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC2, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_uwUniFrac_all$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC2: \", round(100*pcoa_uwUniFrac_all$data$ProportionExplained[2]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-2_uwUniFrac_all.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "#Plot PC1 vs. PC3 (uwUniFrac_all)\n",
                "pcoa_uwUniFrac_all$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC3, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_uwUniFrac_all$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC3: \", round(100*pcoa_uwUniFrac_all$data$ProportionExplained[3]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-3_uwUniFrac_all.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "\n",
                "#Plot PC1 vs. PC2 (wUniFrac_all)\n",
                "pcoa_wUniFrac_all$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC2, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_wUniFrac_all$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC2: \", round(100*pcoa_wUniFrac_all$data$ProportionExplained[2]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-2_wUniFrac_all.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "#Plot PC1 vs. PC3 (wUniFrac_all)\n",
                "pcoa_wUniFrac_all$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC3, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_wUniFrac_all$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC3: \", round(100*pcoa_wUniFrac_all$data$ProportionExplained[3]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-3_wUniFrac_all.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "\n",
                "#################\n",
                "### WILD ONLY ###\n",
                "#################\n",
                "\n",
                "\n",
                "cbp2_beta_wild <- c(\"orange\", \"red\", \"blue\")\n",
                "\n",
                "\n",
                "\n",
                "#Plot PC1 vs. PC2 (uwUniFrac_wild)\n",
                "pcoa_uwUniFrac_wild$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC2, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_uwUniFrac_wild$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC2: \", round(100*pcoa_uwUniFrac_wild$data$ProportionExplained[2]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta_wild) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-2_uwUniFrac_wild.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "#Plot PC1 vs. PC3 (uwUniFrac_wild)\n",
                "pcoa_uwUniFrac_wild$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC3, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_uwUniFrac_wild$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC3: \", round(100*pcoa_uwUniFrac_wild$data$ProportionExplained[3]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta_wild) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-3_uwUniFrac_wild.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "\n",
                "#Plot PC1 vs. PC2 (wUniFrac_wild)\n",
                "pcoa_wUniFrac_wild$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC2, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_wUniFrac_wild$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC2: \", round(100*pcoa_wUniFrac_wild$data$ProportionExplained[2]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta_wild) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-2_wUniFrac_wild.svg\", width = 20, height = 11, dpi = 300)\n",
                "\n",
                "#Plot PC1 vs. PC3 (wUniFrac_wild)\n",
                "pcoa_wUniFrac_wild$data$Vectors %>%\n",
                "  rename(\"X.SampleID\"=SampleID) %>%\n",
                "  left_join(metadata_pcoa) %>%\n",
                "  \n",
                "  ggplot(aes(x=PC1, y=PC3, color=Population)) +\n",
                "  scale_x_reverse() +\n",
                "  geom_point(size=5) +\n",
                "  xlab(paste(\"PC1: \", round(100*pcoa_wUniFrac_wild$data$ProportionExplained[1]), \"%\")) +\n",
                "  ylab(paste(\"PC3: \", round(100*pcoa_wUniFrac_wild$data$ProportionExplained[3]), \"%\")) +\n",
                "  #Custom manual colours\n",
                "  scale_colour_manual(values=cbp2_beta_wild) +\n",
                "  \n",
                "  theme(axis.text.x = element_text(face=\"bold\", size=14), \n",
                "        axis.text.y = element_text(face=\"bold\", size=14),\n",
                "        axis.title.x = element_text(size=20, face=\"bold\"),\n",
                "        axis.title.y = element_text(size=20, face=\"bold\"),\n",
                "        axis.line = element_line(colour = \"black\"),\n",
                "        #Background panel\n",
                "        panel.background = element_rect(fill = \"White\"),\n",
                "        panel.grid.major = element_line(colour = \"white\"), \n",
                "        panel.grid.minor = element_line(colour = \"white\"),\n",
                "        #Legend\n",
                "        legend.title = element_text(size=0),\n",
                "        legend.text = element_text(size=16),\n",
                "        legend.key = element_rect(fill = \"white\", color = NA),\n",
                "        legend.key.size = unit(3.5, \"line\"))\n",
                "\n",
                "#Save image\n",
                "ggsave(filename = \"PC1-3_wUniFrac_wild.svg\", width = 20, height = 11, dpi = 300)\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "## Venn diagrams (figures 3 and 5)\n",
                "\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "##Captive vs. Wild\n",
                "#Store only row names (ASV names) as character vectors\n",
                "captive.asv.names <- row.names(captive.asvs)\n",
                "wild.asv.names <- row.names(wild.asvs)\n",
                "\n",
                "captive.vs.wild <- list(Captive = captive.asv.names, Wild = wild.asv.names)\n",
                "\n",
                "venn.diagram(captive.vs.wild, print.mode = c(\"raw\",\"percent\"), fill = c(\"black\", \"white\"), cat.fontface = 2, cat.pos = 180, imagetype = \"png\", filename = \"R_figures/captive-vs-wild-venn.png\")\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "![Captive and Wild](./R_figures/captive-vs-wild-venn.png \"Title\")\n",
                "\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "##Wild populations\n",
                "#Store only row names (ASV names) as character vectors\n",
                "brookfield.asv.names <- row.names(brookfield.asvs)\n",
                "kooloola.asv.names <- row.names(kooloola.asvs)\n",
                "wonga.asv.names <- row.names(wonga.asvs)\n",
                "\n",
                "wild.comparison <- list(Brookfield = brookfield.asv.names, Kooloola = kooloola.asv.names, Wonga = wonga.asv.names)\n",
                "\n",
                "venn.diagram(wild.comparison, print.mode = c(\"raw\",\"percent\"), fill = c(\"orange\", \"red\", \"blue\"), cat.fontface = 2, cat.pos = c(320, 40, 180), imagetype = \"png\", filename = \"R_figures/wild-populations-venn.png\")\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "![Captive and Wild](./R_figures/wild-populations-venn.png \"Title\")\n",
                "\n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "sessionInfo()\n",
                "\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "\n",
                "\n"
            ]
        }
    ],
    "metadata": {
        "anaconda-cloud": "",
        "kernelspec": {
            "display_name": "R",
            "langauge": "R",
            "name": "ir"
        },
        "language_info": {
            "codemirror_mode": "r",
            "file_extension": ".r",
            "mimetype": "text/x-r-source",
            "name": "R",
            "pygments_lexer": "r",
            "version": "3.4.1"
        }
    },
    "nbformat": 4,
    "nbformat_minor": 1
}
