# This folder contains the supplementary files and figures for the manuscript
### To view the interactive qiime visualizaton files (.qzv), drag/drop them into the box (or select them) on the https://view.qiime2.org/ website.


-SI_Figure_1:
PCoA plot of unweighted UniFrac distances including the 7 outlier Brookfield samples.

-SI_Figure_2:
Phylum-level taxonomic barplots including the 7 outlier Brookfield samples

-SI_Figure_3:
PCoA plots of weighted UniFrac distances for captive and wild samples

-SI_Figure_4:
PCoA plots of weighted UniFrac distances for the different populations of wild samples.

-SI_Table_1:
Pairwise Kruskal-Wallis test results for Faithâ€™s phylogenetic diversity metric comparisons between populations.

-SI_Table_2:
Pairwise Kruskal-Wallis test results for observed features (richness) metric comparisons between populations.

-SI_Table_3:
SINA alignment results of the most abundant core ASVs in SHNWs.

-SI_Table_4:
SourceTracker2 results testing the estimated proportion of faecal bacteria coming from soil. https://doi.org/10.25909/12979157

-SI_File_1:
QIIME2 qzv file of taxonomic bar plots per sample.

-SI_File_2:
QIIME2 qzv file of ANCOM test at family level (captive and wild samples).

-SI_File_3:
QIIME2 qzv file of ANCOM test at the ASV level (captive and wild samples).

-SI_File_4:
QIIME2 qzv file of ANCOM test at family level (between different wild populations).

-SI_File_5:
QIIME2 qzv file of ANCOM test at the ASV level (between different wild populations).
